src/index.js

```diff
@@ -117,8 +123,8 @@ export const Middleware = {
 
   timestamp: next => msg => next(merge(msg, {
     timestamp: (typeof msg.timestamp === 'Date' ?
-                   (msg.timestamp.getTime() + '000000') :
-                   Date.now() + '000000')
+                   Number((msg.timestamp.getTime() + '000000')) :
```

Comment:
I'm not so sure about this change: 
<img width="266" alt="screen shot 2016-04-27 at 21 36 18" src="https://cloud.githubusercontent.com/assets/193115/14865700/18ef78a2-0cc0-11e6-8241-932ddf78d2e0.png">